include/pybind11/numpy.h

```diff
@@ -342,7 +342,7 @@ class unchecked_reference {
      * dimensionality, this is not checked (and so is up to the caller to use safely).
      */
     template <typename... Ix> const T &operator()(Ix... index) const {
-        static_assert(sizeof...(Ix) == Dims || Dynamic,
+        static_assert((ssize_t) sizeof...(Ix) == Dims || Dynamic,
```

Comment:
`ssize_t{x}` is a nice and safe alternative to C-style casts when `x` is a compile-time constant.